docs/source/libmav/libmav.py

```diff
@@ -0,0 +1,1046 @@
+        """
+        pass
+
+    
+    
+class TCPServer():
+    """Represents a physical connection for listening on a specified local TCPIP port for MAVLink traffic.
+
+    An object of this type may be passed to `NetworkRuntime` as shown.
+
+    ```python
+    connection_port = 5760
+    physical_interface = libmav.TCPServer(connection_port)
+
+    # Create the runtime object
+    connection_runtime = libmav.NetworkRuntime(common_messages, heartbeat_message, physical_interface) 
+    ```
+
+    Note that multiple remote computers might connect to this interface.
+    A separate `ConnectionPartner` instance is created for each, and made available to user code in the `NetworkRuntime.on_connection()` callback.
+    """
+
+
+    def __init__(local_port):
+        """Object initialiser
+        
+        Args:
+            local_port (int): The port on local computer to connect to. 
+        """
+        pass    
+         
+    
+    def close(self):
+        """Close the current physical connection.
+        
+        Close the socket connection and clean up any `ConnectionParnter` instances associated with it.
+        """
+        pass
+    
+
+class UDPClient():
+    """Represents a UDP socket connection to a port on a remote computer.
+    
+    This type of object is used to connect to remote MAVLink connection partner that is acting as a UDP server, and listening for connections.
+    The client must send a heartbeat, as the server is usually not initially sending MAVLink packets on the interface.
+
+    An object of this type may be passed to `NetworkRuntime`.
+
+    """
+
+
+    def __init__(remote_address, remote_port):
+        """Construct the UDPClient object, specifing the remote address and port.
+
+        Args:
+            remote_address (string): The IP address of the remote computer to connect to.
+            remote_port (int): The port on remote computer to connect to. 
+        """
+        pass
+
+    
+    def close(self):
+        """Close the UDP socket and clean up the (single) associated `ConnectionPartner` instance.
+        """
+        pass
+
+
+class UDPServer():
+    """Represents a physical connection for listening on a specified local UDP port for MAVLink traffic.
+
+    An object of this type may be passed to `NetworkRuntime` as shown.
+
+    ```python
+    # Create the a UDP server interface object that listens for MAVLink traffic.
+    connection_port = 14551
+    physical_interface = libmav.UDPServer(connection_port)
+    ```
+
+    Note that multiple remote computers might connect to this interface, and `ConnectionPartner` would be created for each and called back using 
+    A separate `ConnectionPartner` instance is created for each, and made available to user code in the `NetworkRuntime.on_connection()` callback.
+    <!-- what happens if you don't register for the callback? Is the object created but inaccessible? -->
```

Comment:
what happens if you don't register for the callback? Is the Connection object created but inaccessible?
```

Reply:
Correct - the first connection you'll always get with the `awaitConnection` call, but subsequent connections you wouldn't, so it is only relevant for server type usage.
With the reusable connections overhaul in the C++ side, I think I can then add a method `getConnections()` or similar, to always be able to retrieve the list of currently active connections.